connection_pool/connector_test.go

```diff
@@ -0,0 +1,1168 @@
+package connection_pool_test
```

Comment:
Was the test built from scratch or was it based on some existing test suite?

Reply:
No, it was done by my hands.